src/systimer_posix.c

```diff
@@ -0,0 +1,102 @@
+/* Could live without sigaction but it seems to be pretty universal */
+
+/* This uses the POSIX 2001 API
+   which allows per-thread timing and minimises risk of
+   interference with other code using timers.
+
+   Sadly it's not always available, so we have an alternative implementation
+   below using the odler setitimer interface */
+
+/* Handler for the Alarm signal */
+
+int SyHaveAlarms = 1;
+
+/* This API lets us pick wich signal to use */
+#define TIMER_SIGNAL SIGVTALRM
+
+
+/* For now anyway we create one timer at initialisation and use it */
+static timer_t syTimer = 0;
+
+#if defined(SYS_IS_CYGWIN32) || defined(SYS_IS_FREEBSD)
+#define MY_CLOCK CLOCK_REALTIME
+#else
+#define MY_CLOCK CLOCK_THREAD_CPUTIME_ID
+#endif
+
+static void SyInitAlarm( void ) {
+/* Create the CPU timer used for timeouts */
+  struct sigevent se;
+  se.sigev_notify = SIGEV_SIGNAL;
+  se.sigev_signo = TIMER_SIGNAL;
+  se.sigev_value.sival_int = 0x12345678;
+  if (timer_create( MY_CLOCK, &se, &syTimer)) {
+    Pr("#E  Could not create interval timer. Timeouts will not be supported\n",0L,0L);
+    SyHaveAlarms = 0;
+  }
+}
+
+static void syAnswerAlarm ( int signr, siginfo_t * si, void *context)
+{
+    /* interrupt the executor
+       Later we might want to do something cleverer with throwing an
```

Comment:
No. I meant that this code could possibly be improved. So later in development time, not execution time